manageiq-operator/internal/controller/manageiq_controller.go

```diff
@@ -271,8 +273,21 @@ func (r *ManageIQReconciler) SetupWithManager(mgr ctrl.Manager) error {
 				}
 			}
 			return reconcileRequests
-		})).
-		Complete(r)
+		}))
+
+	if cfg, err := config.GetConfig(); err == nil {
+		if client, err := client.New(cfg, client.Options{Scheme: r.Scheme}); err == nil {
+			// Watch Routes if we are running in OpenShift
+			if err := client.List(context.TODO(), &routev1.RouteList{}); err == nil {
+				logger.Info("Adding watch for Routes!")
+				controller = controller.Owns(&routev1.Route{})
+			} else {
+				logger.Info(fmt.Sprintf("Skipping watch for Routes! %s", err))
+			}
```

Comment:
Should this conditional also be on the other 3 `if` clauses if they fail?

Reply:
I don't think those would fail, but I added some anyway.